src/shadow-embed.js

```diff
@@ -78,13 +81,24 @@ export function createShadowRoot(hostElement) {
         },
       });
     }
-    return shadowRoot;
   } else if (shadowDomSupported == ShadowDomVersion.V0) {
-    return hostElement.createShadowRoot();
+    shadowRoot = hostElement.createShadowRoot();
+  } else {
+    shadowRoot = createShadowRootPolyfill(hostElement);
+  }
+
+  if (!isShadowCssSupported()) {
```

Comment:
This branch should be included in tests.

Reply:
Cool, I can do that. I order to do checking of the class list, and trigger this branch, I added `toArray()` and sinon for stubbing in the test. Is that alright? I'm debugging some syntax, and then had to run to class.

Reply:
We usually implement `setFeatureForTesting`, like this: https://github.com/ampproject/amphtml/blob/master/src/web-components.js#L38

That's very simple since the compiler removes all "ForTesting" and optimizes this out completely.

Reply:
Ahhh so I was having trouble using arrow functions with parameters in the test. And I don't think that the browserify supports arrow functions with parameters.

Since I am trying to use `array.some()`

Reply:
Awesome, I'll look into that, thanks!

Reply:
Done. I wrote tests to ensure the Class and Id were applied, and that the css was transformed correctly :)